src/shell.js

```diff
@@ -218,7 +218,7 @@ if (ENVIRONMENT_IS_NODE) {
 
 #include "node_shell_read.js"
 
-  if (process.argv.length > 1) {
+  if (!Module['thisProgram'] && process.argv.length > 1) {
```

Comment:
Is this related to this PR?

Reply:
Yes, one of the tests here verifies the contents of arg0.   

This fixes a (what I claim is a) bug where were not honoring the existing `{{{ makeModuleReceive('thisProgram') }}}` under node.